index.js

```diff
@@ -48,6 +55,201 @@ class ExecutorQueue extends Executor {
+        this.scheduler.on('transferred_job', (timestamp, job) =>
+            winston.info(`scheduler enqueuing job timestamp  >>  ${JSON.stringify(job)}`));
+
+        process.on('SIGTERM', () => {
+            this.multiWorker.end((error) => {
+                if (error) {
+                    winston.error(`failed to end the worker: ${error}`);
+                }
+
+                this.scheduler.end((err) => {
+                    if (err) {
+                        winston.error(`failed to end the scheduler: ${err}`);
+                        process.exit(128);
+                    }
+                    process.exit(0);
+                });
+            });
+        });
+    }
+
+    /**
+     * Posts a new build event to the API
+     * @method postBuildEvent
+     * @param {Object} config          Configuration
+     * @param {Object} config.pipeline Pipeline of the job
+     * @param {Object} config.job      Job object to create periodic builds for
+     * @return {Promise}
+     */
+    async postBuildEvent({ pipeline, job }) {
+        const jwt = this.tokenGen(pipeline.admins[0], {}, pipeline.scmContext);
+
+        const options = {
+            url: '/events',
+            method: 'POST',
+            headers: {
+                Authorization: `Bearer ${jwt}`,
+                'Content-Type': 'application/json'
+            },
+            body: {
+                pipelineId: pipeline.id,
+                startFrom: job.name
+            }
+        };
+
+        return req(options, (err, response) => {
+            if (!err && response.statusCode === 200) {
+                return Promise.resolve(response);
+            }
+
+            return Promise.reject(err);
+        });
+    }
+
+    /**
+     * Starts a new periodic build in an executor
+     * @method _startPeriodic
+     * @param {Object}   config             Configuration
+     * @param {Object}   config.pipeline    Pipeline of the job
+     * @param {Object}   config.job         Job object to create periodic builds for
+     * @param {Function} config.tokenGen    Function to generate JWT from username, scope and scmContext
+     * @param {Boolean}  config.isUpdate    Boolean to determine if updating existing periodic build
+     * @return {Promise}
+     */
+    async _startPeriodic(config, triggerBuild = false) {
+        // eslint-disable-next-line max-len
+        const buildCron = hoek.reach(config.job, 'permutations>0>annotations>screwdriver.cd/buildPeriodically',
+            { separator: '>' });
+
+        // Save tokenGen to current executor object so we can access it in postBuildEvent
+        if (!this.tokenGen) {
+            this.tokenGen = config.tokenGen;
+        }
+
+        if (config.isUpdate) {
+            // eslint-disable-next-line no-underscore-dangle
+            await this._stopPeriodic({
+                jobId: config.job.id
+            });
+            delete config.isUpdate;
```

Comment:
Why do we need this `delete`? Also, generally we don't want to modify the param.

Reply:
Changed to use Object.assign